test_conformance/subgroups/test_workgroup_non_uniform_vote.cpp

```diff
@@ -0,0 +1,344 @@
+//
+// Copyright (c) 2020 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "procs.h"
+#include "subhelpers.h"
+#include "harness/typeWrappers.h"
+
+template <typename Ty, int Which> struct AAN_OPERATION;
+
+template <typename Ty> struct AAN_OPERATION<Ty, 0>
+{
+    static bool calculate(const Ty *mx)
+    {
+        bool taa = false;
+        for (int i = 0; i < NON_UNIFORM; ++i)
+            taa |= mx[i]
+                != 0; // return non zero if value non zero at least for one
+        return taa;
+    }
+};
+
+template <typename Ty> struct AAN_OPERATION<Ty, 1>
+{
+    static bool calculate(const Ty *mx)
+    {
+        bool taa = true;
+        for (int i = 0; i < NON_UNIFORM; ++i)
+            taa &= mx[i] != 0; // return non zero if value non zero for all
+        return taa;
+    }
+};
+
+template <typename Ty> struct AAN_OPERATION<Ty, 2>
+{
+    static bool calculate(const Ty *mx)
+    {
+        bool taa = 1;
+        for (int i = 0; i < NON_UNIFORM; ++i)
+            taa &= compare_ordered(mx[i],
+                                   mx[0]); // return non zero if all the same
+        return taa;
+    }
+};
+
+// DESCRIPTION :
+// Test any/all/all_equal non uniform test functions non uniform:
+// Which 0 - any, 1 - all, 2 - all equal
+template <typename Ty, int Which> struct AAN
+{
+    static void gen(Ty *x, Ty *t, cl_int *m, int ns, int nw, int ng)
+    {
+        int i, ii, j, k, n;
```

Comment:
These are C89 style declarations - fold these to their initialisation points, the for-loops. Restrict their scope.

Reply:
I think such change is not important now